### PR TITLE
Feature: ZEN-34387

### PR DESCRIPTION
**Pull request by zensaiello**

### CODE REVIEW

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

File: Products/ZenEvents/TrapFilter.py [hunk 1]
```diff
@@ -240,22 +287,24 @@ def _parseV1FilterDefinition(self, lineNumber, action, remainingTokens):
...
             self._v1Filters[filterDef.levels()] = filtersByLevel
-        elif key not in filtersByLevel:
-            filtersByLevel[key] = filterDef
-        else:
+        elif key in filtersByLevel:
             previousDefinition = filtersByLevel[key]
-            return "OID '%s' conflicts with previous definition at line %d" % (oid, previousDefinition.lineNumber)
+            if self._parseDefConflictShouldOverride(filterDef, previousDefinition) is False:
+                return "OID '%s' conflicts with previous definition at line %d" % (oid, previousDefinition.lineNumber)
+            filtersByLevel[key] = filterDef
```
Comment (StevePC) [inline Products/ZenEvents/TrapFilter.py:294]: need to unindent to preserve logic

File: Products/ZenEvents/TrapFilter.py [hunk 2]
```diff
@@ -276,17 +325,17 @@ def _parseV2FilterDefinition(self, lineNumber, action, remainingTokens):
...
             self._v2Filters[filterDef.levels()] = filtersByLevel
-        elif oid not in filtersByLevel:
-            filtersByLevel[oid] = filterDef
-        else:
+        elif oid in filtersByLevel:
             previousDefinition = filtersByLevel[oid]
-            return "OID '%s' conflicts with previous definition at line %d" % (oid, previousDefinition.lineNumber)
+            if self._parseDefConflictShouldOverride(filterDef, previousDefinition) is False:
+                return "OID '%s' conflicts with previous definition at line %d" % (oid, previousDefinition.lineNumber)
+            filtersByLevel[oid] = filterDef
```
Comment (StevePC) [inline Products/ZenEvents/TrapFilter.py:338]: unindent as well

**Review by StevePC — CHANGES_REQUESTED** [review]

File: Products/ZenModel/migrate/zentrapSvcDefForFiltering.py [hunk 3]
```diff
@@ -52,22 +49,137 @@ def cutover(self, dmd):
...
+                log.error(
+                    "%s %s service: No 'zentrap.filter.conf' File configuration found; "
+                    "broken service def; skipping.", collectorName, svc.name)
+                continue
+
+            ofc = next(
+                (x for x in svc.originalConfigs if x.name == '/opt/zenoss/etc/zentrap.filter.conf'),
+                None
+            )
+            if fc.content == ofc.content:
```
Comment (StevePC) [inline Products/ZenModel/migrate/zentrapSvcDefForFiltering.py:68]: unlikely but ofc could be None here

**Review by StevePC — CHANGES_REQUESTED** [review]

**Review by StevePC — APPROVED** [review]

**Comment by Deer-WarLord** [comment]
1. If we're applying a new configuration to the services, shouldn't that also be defined in the zenoss-service repo?
2. Maybe it's worth applying the new graph config through the JSON file instead of direct [Python objects configuration](https://github.com/zenoss/zenoss-prodbin/pull/4238/files#diff-1e95c880eaef298fa9a2232b676022018f4771c595fb4ffe43dacd47bc0f1e5fR125-R171)?

**Review by Deer-WarLord — APPROVED** [review]

**Comment by jpeacock-zenoss** [comment]
please test this

**Pull request merged**